File: Leetcode/2210 - Count Hills and Valleys in an Array/solution.ts
function countHillValley(nums: number[]): number {
    const n = nums.length;
    let i = 1, count = 0;

    while (i < n - 1) {
        let j = i - 1;
        let k = i + 1;
        while (j > 0 && nums[j] === nums[i]) j--;
        while (k < n - 1 && nums[k] === nums[i]) k++;

        if ((nums[j] < nums[i] && nums[i] > nums[k]) ||
            (nums[j] > nums[i] && nums[i] < nums[k])) {
            count++;
        }
        i = k;
    }

    return count;
};

File: Leetcode/2210 - Count Hills and Valleys in an Array/solution.py
def countHillValley(self, nums: List[int]) -> int:
    n = len(nums)
    i = 1
    count = 0

    while i < n - 1:
        j, k = i - 1, i + 1
        while j > 0 and nums[j] == nums[i]:
            j -= 1
        while k < n - 1 and nums[k] == nums[i]:
            k += 1

        if (nums[j] < nums[i] and nums[i] > nums[k]) or (nums[j] > nums[i] and nums[i] < nums[k]):
            count += 1
        i = k

    return count